sjsonnet/src-jvm/sjsonnet/Platform.scala

```diff
@@ -45,4 +56,78 @@ object Platform {
       .map{ b => String.format("%02x", new java.lang.Integer(b & 0xff))}
       .mkString
   }
+
+  /**
+   * Compacts a [[LinkedHashMap]] as small as possible. There are 3 different cases:
+   * 1. If the map is empty, return [[Collections.emptyMap]]. This is an immutable singleton - and
+   *    thus is space efficient.
+   * 2. If the map has 1 element, so we return an immutable single element map.
+   * 3. Otherwise, we return a new [[LinkedHashMap]] with the same contents as the original map
+   *    that's been compacted to the smallest size possible - subject to the desired load factor:
+   *    (num elements) / load-factor + 1. The default load factor is 0.75.
+   *
+   * Return types from (1) & (2) are [[java.util.Map]] - where as 3 is a [[LinkedHashMap]]. So we
+   * need to pick the common type - which is [[java.util.Map]].
+   *
+   * All returned maps preserve the insertion order of the original map. No map returned from this
+   * method should be mutated.
+   */
+  /*
+  def compactHashMap[K, V: TypeTag](map: LinkedHashMap[K, V]): JMap[K, V] = {
```

Comment:
Don't use Scala reflection for simple stuff like this. (It is almost always the wrong choice.) All you need is a `ClassTag` for Java reflection.

Reply:
Yeah, I mostly did this for expediency of testing. I was less concerned with performance and instead wanted to quickly iterate on the heap-usage of the fully processed + optimized AST. The reflection was clearly slowing down the benchmark by 2x.